collector/lib/K8s.h

```diff
@@ -0,0 +1,56 @@
+#ifndef _K8S_H_
+#define _K8S_H_
+
+#include <sstream>
+#include <string_view>
+
+#include "system-inspector/EventExtractor.h"
+
+namespace collector {
+
+class K8s {
+ public:
+  K8s(sinsp* inspector) : inspector_(inspector) {
+    event_extractor_.Init(inspector);
+  }
+
+  inline std::string_view GetNamespace(sinsp_evt* event) {
+    const char* ns = event_extractor_.get_k8s_namespace(event);
+    return ns != nullptr ? ns : "";
+  }
+
+  inline std::string_view GetNamespace(const std::string& container_id) {
+    return GetContainerLabel(container_id, "io.kubernetes.pod.namespace");
+  }
+
+  std::string GetContainerLabels(const std::string& container_id) {
+    const auto container = inspector_->m_container_manager.get_container(container_id);
+    if (container == nullptr) {
+      return "";
+    }
+
+    std::stringstream ss;
+
+    for (const auto& [key, value] : container->m_labels) {
+      ss << key << ":" << value << ",";
+    }
+
+    return ss.str();
+  }
+
+  inline std::string_view GetContainerLabel(const std::string& container_id, const std::string& label) {
+    const auto container = inspector_->m_container_manager.get_container(container_id);
+    if (container == nullptr || container->m_labels.count(label) == 0) {
```

Comment:
Using `count` seems inefficient.

Reply:
It's what Falco does so I didn't pay it too much care, `count` returns either 1 or 0 since `map` doesn't allow duplicate keys (see https://en.cppreference.com/w/cpp/container/map/count)

I could change it to `find`, but I think the slowest operation will be creating the copy of the string either way

(And as I write this comment I realize there are some changes missing here that are in #1614, I'll add them back and rebase the other 2 PRs)